### ISSUE TITLE
Why my finetuned LoRA in my own voice (English) doesn't sound like me

### ISSUE THREAD

**Issue by beinghorizontal** [voicepowered-ai/VibeVoice-finetuning#15]
I have trained on 525 lines of text on my English dataset with my own voice, with no background noise or anything. But still, when I attached the Lora node to the voice node, the output was *that* of *a* completely different person. The weird thing is, when I attached my own voice clip to the clone input, the output was still of a completely different person, which wasn't the case without adding the LoRA node. Here are the details.

My training args

`python -m src.finetune_vibevoice_lora --model_name_or_path E:/comf/ComfyUI/models/vibevoice/VibeVoice-1.5B --train_jsonl D:/anaconda/envs/vibevoice/Scripts/training.jsonl --text_column_name text --audio_column_name audio --voice_prompts_column_name audio --output_dir E:/tts_merged/vibevoice_model --per_device_train_batch_size 1 --gradient_accumulation_steps 1 --learning_rate 2.5e-5 --num_train_epochs 5 --logging_steps 10 --save_steps 500 --eval_steps 500 --report_to tensorboard --remove_unused_columns False --bf16 True --do_train --gradient_checkpointing False --ddpm_batch_mul 4 --diffusion_loss_weight 1.4 --train_diffusion_head True --ce_loss_weight 0.04 --voice_prompt_drop_rate 0.2 --lora_target_modules q_proj,k_proj,v_proj,o_proj,gate_proj,up_proj,down_proj --lr_scheduler_type cosine --warmup_ratio 0.03 --gradient_clipping --max_grad_norm 0.8`

 I tried it both ways, with and without this arg
`voice_prompts_column_name audio`

 These are the last few steps of my fine-tuning training output.

`{'train/ce_loss': 1.2325459718704224, 'train/diffusion_loss': 0.19995176792144775, 'epoch': 5.0}
{'train/learning_rate_real': 3.8064728369313627e-11, 'epoch': 5.0}
{'debug/num_tok_total': 137.0, 'debug/num_tok_loss': 72.0, 'debug/num_lat_total': 137.0, 'debug/num_lat_loss': 72.0, 'epoch': 5.0}
{'train/ce_loss': 2.1148324012756348, 'train/diffusion_loss': 0.21310661733150482, 'epoch': 5.0}
{'train/learning_rate_real': 9.516185714431025e-12, 'epoch': 5.0}
100%|██████████████████████████████████████████████████████████████████████████████| 2625/2625 [25:30<00:00,  1.82it/s]D:\anaconda\envs\vibevoice\Lib\site-packages\peft\utils\save_and_load.py:286: UserWarning: Could not find a config file in  - will assume that the vocabulary was not modified.
  warnings.warn(
{'train_runtime': 1532.0732, 'train_samples_per_second': 1.713, 'train_steps_per_second': 1.713, 'train_loss': 0.5533197813488189, 'epoch': 5.0}
100%|██████████████████████████████████████████████████████████████████████████████| 2625/2625 [25:31<00:00,  1.71it/s]
D:\anaconda\envs\vibevoice\Lib\site-packages\peft\utils\save_and_load.py:286: UserWarning: Could not find a config file in  - will assume that the vocabulary was not modified.
  warnings.warn(`

My dataset is in JSONL format 

Example:

`{"text": "Speaker 1: Combine all the data. Put it all together and save it in local folder", "audio": "J:\\asr_data\\medium_alex\\davinci_wavs3/Recording_3.wav"}
{"text": "Speaker 1: Create the Hugging Face format from my csv file of this path.", "audio": "J:\\asr_data\\medium_alex\\davinci_wavs3/Recording_4.wav"}
{"text": "Speaker 1: Combine old hugging face data with the new one.", "audio": "J:\\asr_data\\medium_alex\\davinci_wavs3/Recording_5.wav"}`

 I confirm that the file path of the audio and the corresponding transcription match.

 ComfyUi setup. I tried both ways, adding my voice clip and without it. And both times, the output was of a different person, which wasn't the case without LoRa. 

<img width="1418" height="863" alt="Image" src="https://github.com/user-attachments/assets/2145c0cd-710f-4139-a789-bf869f30d5f9" />

I think I'm missing something very obvious, and my fine-tuning is actually generalising, instead of fine-tuning on my own voice. What is going on here?

**Comment by jpgallegoar** [comment]
Hello, in order to train a single voice you should set --voice_prompt_drop_rate 1 . This way, the model will learn to generalize on that specific voice only

**Comment by beinghorizontal** [comment]
Thanks for the quick reply! I'll try that right away and report back. Interestingly, I suspected that particular prompt might be the issue, but I even lowered the value to 0.002 and began fine-tuning. I'll go with the value you suggested.

**Comment by jpgallegoar-vpai** [comment]
You're welcome! Best of luck. Also, a few more tips:
1) Your dataset should have around 1-3h minimum of that person speaking, with as much varied content as possible, or at least content that spans the expected generation content / style. If you don't train yourself acting sad, you shouldn't expect the model to be able to copy you acting sad, for example.
2) During inference, it's been reported that unplugging the voice conditioning (voice prompt) while using a unique voice LoRa can improve it's naturalness, as you don't need it anymore since the model is trained specifically for that voice only. As a bonus, disabling voice prompt will make the model generate slightly faster.
3) Train for at least 3 epochs, but perhaps train more while saving all checkpoints and then test them all (I have yet to do this myself so I can't tell you specifically what worked for me)a

**Comment by beinghorizontal** [comment]
@jpgallegoar-vpai Thanks for the bonus tips! But I didn’t quite understand this part:

> During inference, it's been reported that unplugging the voice conditioning (voice prompt) while using a unique voice LoRa can improve its naturalness.

**Comment by jpgallegoar-vpai** [comment]
> [@jpgallegoar-vpai](https://github.com/jpgallegoar-vpai) Thanks for the bonus tips! But I didn’t quite understand this part:
> 
> > During inference, it's been reported that unplugging the voice conditioning (voice prompt) while using a unique voice LoRa can improve its naturalness.

<img width="1418" height="863" alt="Image" src="https://github.com/user-attachments/assets/314c70ea-da0d-4171-a81e-9e510c70a487" />

During inference, if you've trained your LoRa correctly, you're able to unplug (voice_to_clone) and the voice is still the one you trained. This will allow the model to use the generalized voice and not stick to the biases that come from that specific voice prompt. Of course you can still use it if you desire, it's just something else you may want to test.

**Comment by beinghorizontal** [comment]
Got it, thanks. I started the training process, so let's see how it goes. I also asked Copilot about voice prompting in TTS, which brought up this question: Does this model support emotion tags? I created a dataset for Orpheus TTS with emotion tags that matched the appropriate emotions (or at least I think so, lol), and Orpheus learned quite well, but it’s erratic. I read some discussions on Reddit where someone mentioned that they had experimented with these kinds of models in the past, and they tend to ignore emotion tags during training but actually use them during inference.

**Comment by jpgallegoar-vpai** [comment]
unfortunately this model does not support emotion tags out of the box. I'm convinced you can train it to accept them, although that's quite an endeavour. How many hours does your emotional dataset have? How did you tag it?

**Comment by beinghorizontal** [comment]
I created a CSV with tags formatted this way: each line contains a maximum of 15 words.

```
I just <groan> don't understand what is going on.
And <sigh> finally combine this data and save it to the local folder.

```
I wrote a script that processes each line in that CSV, displays the text in the terminal, and allocates time based on normal speech rates, along with a tqdm slider as a time indicator. Once I finish speaking the line, including any emotions indicated by tags, the script saves the transcript and the associated WAV file into their respective folders.

Unfortunately, it's only about an hour long. My finetuning with --voice_prompt_drop_rate 1 just finished, and it seems far too short. The output voice was ~30% close. I didn't experience this issue with Orpheus TTS, and even F5-TTS finetuning worked well.

I'll gather more data and review the results. I’m not giving up that easily, LOL! You can close this issue since I got my answer.

**Comment by jpgallegoar-vpai** [comment]
That seems like a great way of creating data! You might want to create a mini app and publish it :D. Please let me know any other questions you may have. BTW I got insight from someone, 3h of data at 6 epochs worked for them. Also consider adding longer speech utterances, as having them so short might impact long output quality

**Comment by beinghorizontal** [comment]
Lol, I am TTS/ASR nerd. Hey, sorry to bother you again, but I tested finetuning once more, this time using 

`--voice_prompt_drop_rate 2`. 

However, that's not the issue.

The strange thing is, when I connected the Lora and then disconnected it while keeping the random seed constant, I got the exact same output. This suggests that my Lora isn't affecting the model at all.

My file structure

Folder PATH listing
¦   
+---loras
¦   +---lora_v1
¦   ¦   ¦   adapter_config.json
¦   ¦   ¦   adapter_model.safetensors
¦   ¦   ¦   diffusion_head_full.bin
¦   ¦   ¦   README.md
¦   ¦   ¦   
¦   ¦   +---acoustic_connector
¦   ¦   ¦       pytorch_model.bin
¦   ¦   ¦       
¦   ¦   +---diffusion_head
¦   ¦   ¦       config.json
¦   ¦   ¦       diffusion_head_full.bin
¦   ¦   ¦       model.safetensors
¦   ¦   ¦       
¦   ¦   +---semantic_connector
¦   ¦           pytorch_model.bin
¦   ¦           
¦   +---lora_v2
¦   ¦   ¦   adapter_config.json
¦   ¦   ¦   adapter_model.safetensors
¦   ¦   ¦   diffusion_head_full.bin
¦   ¦   ¦   README.md
¦   ¦   ¦   
¦   ¦   +---acoustic_connector
¦   ¦   ¦       pytorch_model.bin
¦   ¦   ¦       
¦   ¦   +---diffusion_head
¦   ¦   ¦       config.json
¦   ¦   ¦       diffusion_head_full.bin
¦   ¦   ¦       model.safetensors
¦   ¦   ¦       
¦   ¦   +---semantic_connector
¦   ¦           pytorch_model.bin
¦   ¦           
¦   +---lora_v3
¦   ¦   ¦   adapter_config.json
¦   ¦   ¦   adapter_model.safetensors
¦   ¦   ¦   diffusion_head_full.bin
¦   ¦   ¦   README.md
¦   ¦   ¦   
¦   ¦   +---acoustic_connector
¦   ¦   ¦       pytorch_model.bin
¦   ¦   ¦       
¦   ¦   +---diffusion_head
¦   ¦   ¦       config.json
¦   ¦   ¦       diffusion_head_full.bin
¦   ¦   ¦       model.safetensors
¦   ¦   ¦       
¦   ¦   +---semantic_connector
¦   ¦           pytorch_model.bin
¦   ¦           
¦   +---lora_v4
¦   ¦   ¦   adapter_config.json
¦   ¦   ¦   adapter_model.safetensors
¦   ¦   ¦   diffusion_head_full.bin
¦   ¦   ¦   README.md
¦   ¦   ¦   
¦   ¦   +---acoustic_connector
¦   ¦   ¦       pytorch_model.bin
¦   ¦   ¦       
¦   ¦   +---diffusion_head
¦   ¦   ¦       config.json
¦   ¦   ¦       diffusion_head_full.bin
¦   ¦   ¦       model.safetensors
¦   ¦   ¦       
¦   ¦   +---semantic_connector
¦   ¦           pytorch_model.bin
¦   ¦           
¦   +---lora_v5
¦       ¦   adapter_config.json
¦       ¦   adapter_model.safetensors
¦       ¦   diffusion_head_full.bin
¦       ¦   README.md
¦       ¦   
¦       +---acoustic_connector
¦       ¦       pytorch_model.bin
¦       ¦       
¦       +---diffusion_head
¦       ¦       config.json
¦       ¦       diffusion_head_full.bin
¦       ¦       model.safetensors
¦       ¦       
¦       +---semantic_connector
¦               pytorch_model.bin
¦               
+---tokenizer
¦       merges.txt
¦       tokenizer.json
¦       tokenizer_config.json
¦       vocab.json
¦       
+---VibeVoice-1.5B
¦       config.json
¦       gitattributes
¦       model-00001-of-00003.safetensors
¦       model-00002-of-00003.safetensors
¦       model-00003-of-00003.safetensors
¦       model.safetensors.index.json
¦       preprocessor_config.json
¦       README.md
¦       
+---VibeVoice-Large-Q4
¦       .gitattributes
¦       config.json
¦       finetune_elise_7b_optimal.sh
¦       generation_config.json
¦       load_quantized_4bit.py
¦       minimal_memory_output.wav
¦       model-00001-of-00002.safetensors
¦       model-00002-of-00002.safetensors
¦       model.safetensors.index.json
¦       preprocessor_config.json
¦       quantization_config.json
¦       QUANTIZATION_README.md
¦       quantize_and_save_vibevoice.py
¦       README.md
¦       test_accurate_vram.py
¦       use_quantized_model.py
¦       vibevoice_7gb_target.py
¦       
+---VibeVoice-Large-Q8
        .gitattributes
        config.json
        generation_config.json
        model-00001-of-00003.safetensors
        model-00002-of-00003.safetensors
        model-00003-of-00003.safetensors
        model.safetensors.index.json
        preprocessor_config.json
        README.md
        
Edit: I am using Enemyx-net's 1.8.0 Comfy custom node

**Comment by jpgallegoar-vpai** [comment]
"--voice_prompt_drop_rate 2." is wrong, that's a value from 0 to 1, it's a percentage.

For the LORA not working problem, please take a look at your console log and look at the issue 

https://github.com/voicepowered-ai/VibeVoice-finetuning/issues/13

Look at the log, if it says "Failed to apply LLM LoRA: 'Qwen2Model' object has no attribute 'prepare_inputs_for_generation'"

Do you have transformers==4.51.3 in the comfy env?

**Comment by beinghorizontal** [comment]
Phew! That was the issue. I had the latest transformer, so I uninstalled it and installed the one you recommended, and now it works. The quality is genuinely good, and while the emotion tags don't work, that's fine. It's expressive enough.

**Comment by jpgallegoar-vpai** [comment]
I'm glad it works! For the emotion tags to work you probably need MUCH more training data, like hundreds to thousands of hours. Best of luck in your future experiments!
